crates/pep508-rs/src/marker.rs

```diff
@@ -967,6 +967,87 @@ impl MarkerTree {
         }
     }
 
+    /// Remove the extras from a marker, returning `None` if the marker tree evaluates to `true`.
```

Comment:
Maybe @BurntSushi should review this marker code.

Reply:
Sure, feel free @BurntSushi.

Reply:
Messaged with Andrew, I want to build on this branch (but not related to the markers) so I'm going to merge and address any review comments post-merge.